src/evaluator/index.js

```diff
@@ -237,43 +253,51 @@ export class Evaluator {
         target = await this.evaluateNode(node.target)
       }
 
-      if (target.type !== 'bytes20' &&
-        target.type !== 'address') {
+      if (target.type !== 'bytes20' && target.type !== 'address') {
         this.panic('Target of call expression was not an address')
-      } else if (!Web3Utils.checkAddressChecksum(target.value)) {
+      } else if (!ethers.utils.getAddress(target.value)) {
         this.panic(`Checksum failed for address "${target.value}"`)
       }
 
       const inputs = await this.evaluateNodes(node.inputs)
       const outputs = node.outputs
-      const selectedReturnValueIndex = outputs.findIndex((output) => output.selected)
+      const selectedReturnValueIndex = outputs.findIndex(
+        (output) => output.selected
+      )
       if (selectedReturnValueIndex === -1) {
-        this.panic(`No selected return value for function call "${node.callee}"`)
+        this.panic(
+          `No selected return value for function call "${node.callee}"`
+        )
       }
       const returnType = outputs[selectedReturnValueIndex].type
 
-      const call = ABI.encodeFunctionCall({
-        name: node.callee,
-        type: 'function',
-
-        inputs: inputs.map(({ type }) => ({
-          type,
-          // web3.js 1.x requires the inputs to have names, otherwise it assumes the type is a tuple
-          // We can remove this if web3.js 1.x fixes this, or we upgrade to a newer major version
-          // For reference: this is the problematic bit in web3.js:
-          // https://github.com/ethereum/web3.js/blob/7d1b0eab31ff6b52c170dedc172decebea0a0217/packages/web3-eth-abi/src/index.js#L110
-          name: 'nonEmptyName'
-        }))
-      }, inputs.map((input) => input.value.toString()))
-
-      return this.eth.call({
+      const ethersInterface = new ethers.utils.Interface([
+        {
+          name: node.callee,
+          type: 'function',
+          inputs: inputs.map(({ type }) => ({
+            type
+          })),
+          outputs: outputs.map(({ type }) => ({
+            type
+          })),
+          stateMutability: 'view'
+        }
+      ])
+
+      const call = ethersInterface.encodeFunctionData(
+        node.callee,
+        inputs.map((input) => input.value.toString())
+      )
+
+      const data = await this.provider.call({
         to: target.value,
         data: call
-      }).then(
-        (data) => ABI.decodeParameters(outputs.map((item) => item.type), data)
-      ).then(
-        (returnData) => new TypedValue(returnType, returnData[selectedReturnValueIndex])
-      )
+      })
+
+      const decodeData = ethersInterface.decodeFunctionResult(node.callee, data)
+
+      return new TypedValue(returnType, decodeData[selectedReturnValueIndex])
```

Comment:
Not opposed to the above and being low-level, but I wonder if creating an ethers contract wouldn't be simpler? Effectively we've created an ABI for the interface, and so it seems like something that could just go into the contract object.